bson/src/test/unit/org/bson/codecs/pojo/PojoCustomTest.java

```diff
@@ -545,15 +548,27 @@ public void testInvalidDiscriminatorInNestedModel() {
                         + "'simple': {'_t': 'FakeModel', 'integerField': 42, 'stringField': 'myString'}}"));
     }
 
+    @Test
+    public void testGenericBaseClass() {
+        CodecRegistry registry = fromProviders(new ValueCodecProvider(), PojoCodecProvider.builder().automatic(true).build());
+
+        ConcreteModel model = new ConcreteModel(new ConcreteField("name1"));
+
+        String json = "{\"_t\": \"org.bson.codecs.pojo.entities.ConcreteModel\", \"field\": {\"name\": \"name1\"}}";
+        roundTrip(PojoCodecProvider.builder().automatic(true), GenericBaseModel.class, model, json);
+    }
+
+
+    // TODO: this now fails with a StackOverflowError, which is not desirable
```

Comment:
This is undesirable, but not sure how to address it.  Perhaps another check to make sure that the actual class being encoding/decoded has no generic parameters?  Where to put that check though...

Reply:
I took some look. The issue is a cyclic dependency was found and ended up with LazyCodec as the codec which will lead to infinitive codec lookup loop. Seems the LazyCodec is mainly for breaking cyclic dependency without full-fledged recursive encoding feature. Given its current limit, seems we could throw another exception once LazyCodec was returned from child reigstry.

Reply:
Right, it's an interaction between LazyPropertyModelCodec and the recursive use of types in GenericTreeModel

Reply:
I've commented in `LazyPropertyModelCodec`